contrib/mpi-proxy-split/mpi-wrappers/mpi_collective_wrappers.cpp

```diff
@@ -273,24 +273,6 @@ USER_DEFINED_WRAPPER(int, Reduce_scatter,
   return twoPhaseCommit(comm, realBarrierCb);
 }
 
-int
-MPI_Alltoall_internal(const void *sendbuf, int sendcount,
-                      MPI_Datatype sendtype, void *recvbuf, int recvcount,
-                      MPI_Datatype recvtype, MPI_Comm comm)
-{
-  int retval;
-  DMTCP_PLUGIN_DISABLE_CKPT();
-  MPI_Comm realComm = VIRTUAL_TO_REAL_COMM(comm);
-  MPI_Datatype realSendType = VIRTUAL_TO_REAL_TYPE(sendtype);
-  MPI_Datatype realRecvType = VIRTUAL_TO_REAL_TYPE(recvtype);
-  JUMP_TO_LOWER_HALF(lh_info.fsaddr);
-  retval = NEXT_FUNC(Alltoall)(sendbuf, sendcount, realSendType, recvbuf,
-      recvcount, realRecvType, realComm);
-  RETURN_TO_UPPER_HALF();
-  DMTCP_PLUGIN_ENABLE_CKPT();
-  return retval;
-}
-
```

Comment:
It will be called by `Alltoall`, so it won't compile when we want to use collective version.

BTW, are we plan to eventually merge this branch to master, and use a MACRO switch to control it?

Reply:
Yes, as @jungan writes, "we plan to eventually merge this branch to master, and use a MACRO switch to control it."

The MACRO switch is `MPI_COLLECTIVE_P2P` in `mpi_collective_wrappers.cpp`.  When we merge this into master, we should add a comment in a Makefile that can define `MPI_COLLECTIVE_P2P`, but it should be undefined by default.

@xuyao0127 says that when testing on VASP, this doesn't yet restart.  I plan to work with him this afternoon to try to fix that.  Once we fix that, we can then do another PR.  And once that PR is accepted, we can then merge this into master (cherry-pick the commits from this feature branch, as a new PR against master).